File: java-sdk/radar-schemas-core/src/main/java/org/radarbase/schema/specification/AppDataTopic.java
package org.radarbase.schema.specification;

import static org.radarbase.schema.util.SchemaUtils.expandClass;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import java.util.Map;

@JsonInclude(Include.NON_NULL)
public class AppDataTopic extends DataTopic {
    @JsonProperty("app_provider")
    private String appProvider;

    @JsonSetter
    @SuppressWarnings("PMD.UnusedPrivateMethod")
    private void setAppProvider(String provider) {
        this.appProvider = expandClass(provider);
    }

    public String getAppProvider() {
        return appProvider;
    }

    @Override
    protected void propertiesMap(Map<String, Object> map, boolean reduced) {
        map.put("app_provider", appProvider);
        super.propertiesMap(map, reduced);
    }

    public static class DataField {
        @JsonProperty
        private String name;

        public String getName() {
            return name;
        }
    }
}

File: java-sdk/radar-schemas-core/src/main/java/org/radarbase/schema/specification/AppDataTopic.kt
package org.radarbase.schema.specification

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonSetter
import org.radarbase.config.OpenConfig
import org.radarbase.schema.util.SchemaUtils

@JsonInclude(NON_NULL)
@OpenConfig
class AppDataTopic : DataTopic() {
    @JsonProperty("app_provider")
    @set:JsonSetter
    var appProvider: String? = null
        set(value) {
            field = SchemaUtils.expandClass(value)
        }

    override fun propertiesMap(map: MutableMap<String, Any?>, reduced: Boolean) {
        map["app_provider"] = appProvider
        super.propertiesMap(map, reduced)
    }

    class DataField {
        @JsonProperty
        var name: String? = null
    }
}
